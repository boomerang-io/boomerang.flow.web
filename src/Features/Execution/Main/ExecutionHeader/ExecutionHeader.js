import React from "react";
import PropTypes from "prop-types";
import { withRouter, Link } from "react-router-dom";
import { SkeletonPlaceholder } from "@boomerang-io/carbon-addons-boomerang-react";
import { appLink } from "Config/appConfig";
import FeatureHeader from "Components/FeatureHeader";
import moment from "moment";
import { QueryStatus } from "Constants";
import styles from "./executionHeader.module.scss";

ExecutionHeader.propTypes = {
  workflow: PropTypes.object.isRequired,
  workflowExecution: PropTypes.object.isRequired,
};

function ExecutionHeader({ history, workflow, workflowExecution }) {
  const { state } = history.location;
  const { teamName, initiatedByUserName, trigger, creationDate } = workflowExecution.data;

  return (
    <FeatureHeader includeBorder>
      <div className={styles.container}>
        <section>
          <div className={styles.subtitle}>
            <Link className={styles.activityLink} to={state ? state.fromUrl : appLink.activity()}>
              {state ? state.fromText : "Activity"}
            </Link>
            <p style={{ margin: "0 0.5rem" }}>/</p>
            {!workflow?.data?.name ? (
              <SkeletonPlaceholder className={styles.workflowNameSkeleton} />
            ) : (
              <p>{workflow.data.name}</p>
            )}
          </div>
          <h1 className={styles.title}>Workflow run detail</h1>
        </section>
        {workflowExecution.status === QueryStatus.Success ? (
          <div className={styles.content}>
            <dl className={styles.data}>
              <dt className={styles.dataTitle}>Team</dt>
              <dd className={styles.dataValue}>{teamName}</dd>
            </dl>
            <dl className={styles.data}>
              <dt className={styles.dataTitle}>Initiated by</dt>
              {initiatedByUserName ? (
                <dd className={styles.dataValue}>{initiatedByUserName}</dd>
              ) : (
                <dd aria-label="robot" aria-hidden={false} role="img">
                  {"ðŸ¤–"}
                </dd>
              )}
            </dl>
            <dl className={styles.data}>
              <dt className={styles.dataTitle}>Trigger</dt>
              <dd className={styles.dataValue}>{trigger}</dd>
            </dl>
            <dl className={styles.data}>
              <dt className={styles.dataTitle}>Start time</dt>
              <dd className={styles.dataValue}>{moment(creationDate).format("YYYY-MM-DD hh:mm A")}</dd>
            </dl>
          </div>
        ) : (
          <SkeletonPlaceholder className={styles.headerContentSkeleton} />
        )}
      </div>
    </FeatureHeader>
  );
}

export default withRouter(ExecutionHeader);
