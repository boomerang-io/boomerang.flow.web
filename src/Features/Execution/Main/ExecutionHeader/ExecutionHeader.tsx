import React from "react";
import { Link, useHistory } from "react-router-dom";
import { SkeletonPlaceholder } from "@boomerang-io/carbon-addons-boomerang-react";
import { appLink } from "Config/appConfig";
import FeatureHeader from "Components/FeatureHeader";
import moment from "moment";
import { QueryStatus } from "Constants";
import { WorkflowSummary, WorkflowDetailedActivityExecution } from "Types";
import styles from "./executionHeader.module.scss";

interface HeaderProps {
  workflow: {
    data: WorkflowSummary;
    status: string;
    error: any;
  };
  workflowExecution: {
    data: WorkflowDetailedActivityExecution;
    status: string;
    error: any;
  };
}

const ExecutionHeader: React.FC<HeaderProps> = ({ workflow, workflowExecution }) => {
  const history = useHistory();
  const { state } = history.location;

  const { teamName, initiatedByUserName, trigger, creationDate } = workflowExecution.data;

  //@ts-ignore
  const fromUrl = state ? state.fromUrl : appLink.activity();
  //@ts-ignore
  const fromText = state ? state.fromText : "Activity";

  return (
    <FeatureHeader includeBorder>
      <div className={styles.container}>
        <section>
          <div className={styles.subtitle}>
            <Link className={styles.activityLink} to={fromUrl}>
              {fromText}
            </Link>
            <p style={{ margin: "0 0.5rem" }}>/</p>
            {!workflow?.data?.name ? (
              <SkeletonPlaceholder className={styles.workflowNameSkeleton} />
            ) : (
              <p>{workflow.data.name}</p>
            )}
          </div>
          <h1 className={styles.title}>Workflow run detail</h1>
        </section>
        {workflowExecution.status === QueryStatus.Success ? (
          <div className={styles.content}>
            <dl className={styles.data}>
              <dt className={styles.dataTitle}>Team</dt>
              <dd className={styles.dataValue}>{teamName}</dd>
            </dl>
            <dl className={styles.data}>
              <dt className={styles.dataTitle}>Initiated by</dt>
              {initiatedByUserName ? (
                <dd className={styles.dataValue}>{initiatedByUserName}</dd>
              ) : (
                <dd aria-label="robot" aria-hidden={false} role="img">
                  {"ðŸ¤–"}
                </dd>
              )}
            </dl>
            <dl className={styles.data}>
              <dt className={styles.dataTitle}>Trigger</dt>
              <dd className={styles.dataValue}>{trigger}</dd>
            </dl>
            <dl className={styles.data}>
              <dt className={styles.dataTitle}>Start time</dt>
              <dd className={styles.dataValue}>{moment(creationDate).format("YYYY-MM-DD hh:mm A")}</dd>
            </dl>
          </div>
        ) : (
          <SkeletonPlaceholder className={styles.headerContentSkeleton} />
        )}
      </div>
    </FeatureHeader>
  );
};

export default ExecutionHeader;
