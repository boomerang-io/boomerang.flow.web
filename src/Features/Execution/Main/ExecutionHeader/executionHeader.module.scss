@import "Styles/mixins";

.container {
  h1 {
    margin: 0;
  }
}

.workflowNameSkeleton {
  height: 1.125rem;
}

.content {
  align-items: center;
  display: flex;
  margin: 1.125rem 0 1.5rem 0;
}

.data {
  color: var(--cds-text-01);
  margin-left: 2.5rem;
}

.dataTitle {
  font-size: 0.875rem;
  font-weight: 300;
  letter-spacing: 0;
  line-height: 1.125rem;
  margin-bottom: 0.25rem;
}

.dataValue {
  font-size: 1rem;
  letter-spacing: 0;
  line-height: 1.25rem;
  max-width: 20rem;
  text-transform: capitalize;
  @include text-overflow-ellipsis();
}

.headerContentSkeleton {
  height: 4.31rem;
  width: 40rem;
}
