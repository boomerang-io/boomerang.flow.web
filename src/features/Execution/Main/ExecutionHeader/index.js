import React from "react";
import PropTypes from "prop-types";
import { withRouter, Link } from "react-router-dom";
import { SkeletonPlaceholder } from "carbon-components-react";
import FeatureHeader from "Components/FeatureHeader";
import moment from "moment";
import { REQUEST_STATUSES } from "Config/servicesConfig";
import styles from "./executionHeader.module.scss";

ExecutionHeader.propTypes = {
  workflow: PropTypes.object.isRequired,
  workflowExecution: PropTypes.object.isRequired
};

function ExecutionHeader({ history, workflow, workflowExecution }) {
  const { state } = history.location;
  const { teamName, initiatedByUserName, trigger, creationDate } = workflowExecution.data;

  return (
    <FeatureHeader includeBorder>
      <div className={styles.container}>
        <section>
          <div className={styles.subtitle}>
            <Link to={state ? state.fromUrl : "/activity"}>{state ? state.fromText : "Activity"}</Link>
            <p style={{ margin: "0 0.5rem" }}>/</p>
            {workflow.isFetching ? (
              <SkeletonPlaceholder className={styles.workflowNameSkeleton} />
            ) : (
              <p>{workflow.data.name}</p>
            )}
          </div>
          <h1 className={styles.title}>Workflow run detail</h1>
        </section>
        {workflowExecution.status === REQUEST_STATUSES.SUCCESS ? (
          <div className={styles.content}>
            <div className={styles.data}>
              <p className={styles.dataTitle}>Team</p>
              <p className={styles.dataValue}>{teamName}</p>
            </div>
            <div className={styles.data}>
              <p className={styles.dataTitle}>Initiated by</p>
              {initiatedByUserName ? (
                <p className={styles.dataValue}>{initiatedByUserName}</p>
              ) : (
                <span aria-label="robot" aria-hidden={false} role="img">
                  {"ðŸ¤–"}
                </span>
              )}
            </div>
            <div className={styles.data}>
              <p className={styles.dataTitle}>Trigger</p>
              <p className={styles.dataValue}>{trigger}</p>
            </div>
            <div className={styles.data}>
              <p className={styles.dataTitle}>Team</p>
              <p className={styles.dataValue}>{moment(creationDate).format("YYYY-MM-DD hh:mm A")}</p>
            </div>
          </div>
        ) : (
          <SkeletonPlaceholder className={styles.headerContentSkeleton} />
        )}
      </div>
    </FeatureHeader>
  );
}

export default withRouter(ExecutionHeader);
