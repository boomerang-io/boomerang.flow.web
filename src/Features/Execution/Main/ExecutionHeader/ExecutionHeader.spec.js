import React from "react";
import ExecutionHeader from "./index";
import { queryCaches } from "react-query";

const props = {
  workflowExecution: {
    status: "success",
    data: {
      teamName: "CAI Offerings",
      initiatedByUserName: "Tim Bula",
      trigger: "manual",
      creationDate: "2019-09-03T15:00:00.049+0000",
    },
  },
  workflow: {
    isFetching: false,
    data: {
      name: "Sparkle Flow with extra glitter and donuts on the side",
    },
  },
};

afterEach(() => {
  queryCaches.forEach((queryCache) => queryCache.clear());
});

describe("ExecutionHeader --- Snapshot", () => {
  it("Capturing Snapshot of ExecutionHeader", () => {
    const { baseElement } = rtlRouterRender(<ExecutionHeader {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
