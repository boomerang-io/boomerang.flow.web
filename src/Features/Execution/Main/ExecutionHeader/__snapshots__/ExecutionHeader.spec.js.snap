// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExecutionHeader --- Snapshot Capturing Snapshot of ExecutionHeader 1`] = `
<body
  id="app"
>
  <div>
    <header
      class="bx--bmrg-feature-header container --bordered"
    >
      <section
        class="bx--bmrg-feature-header__content"
      >
        <div
          class="bx--bmrg-feature-header__nav"
        >
          <nav
            aria-label="Breadcrumb"
          >
            <ol
              class="bx--breadcrumb bx--breadcrumb--no-trailing-slash"
            >
              <li
                class="bx--breadcrumb-item"
              >
                <a
                  class="bx--link"
                  href="/activity"
                >
                  Activity
                </a>
              </li>
              <li
                class="bx--breadcrumb-item bx--breadcrumb-item--current"
              >
                <p
                  class="bx--link"
                >
                  Sparkle Flow with extra glitter and donuts on the side
                </p>
              </li>
            </ol>
          </nav>
        </div>
        <hgroup
          class="bx--bmrg-feature-header__header"
        >
          <h1
            class="bx--bmrg-feature-header-text"
          >
            Workflow run detail
          </h1>
        </hgroup>
      </section>
      <div
        class="content"
      >
        <dl
          class="data"
        >
          <dt
            class="dataTitle"
          >
            Team
          </dt>
          <dd
            class="dataValue"
          >
            CAI Offerings
          </dd>
        </dl>
        <dl
          class="data"
        >
          <dt
            class="dataTitle"
          >
            Initiated by
          </dt>
          <dd
            class="dataValue"
          >
            Tim Bula
          </dd>
        </dl>
        <dl
          class="data"
        >
          <dt
            class="dataTitle"
          >
            Trigger
          </dt>
          <dd
            class="dataValue"
          >
            manual
          </dd>
        </dl>
        <dl
          class="data"
        >
          <dt
            class="dataTitle"
          >
            Start time
          </dt>
          <dd
            class="dataValue"
          >
            2019-01-01 12:00 AM
          </dd>
        </dl>
      </div>
    </header>
  </div>
</body>
`;
